workflows/temporal/temporal.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package temporal
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/sdk/client"
+
+	"github.com/dapr/components-contrib/workflows"
+	"github.com/dapr/kit/logger"
+)
+
+// Placeholder string for the task queue
+const TaskQueueString = "TestTaskQueue"
+
+type TemporalWF struct {
+	client client.Client
+	logger logger.Logger
+}
+
+type temporalMetadata struct {
+	Identity string `json:"identity"`
+	HostPort string `json:"hostport"`
+}
+
+// NewTemporalWorkflow returns a new workflow.
+func NewTemporalWorkflow(logger logger.Logger) workflows.Workflow {
+	s := &TemporalWF{
+		logger: logger,
+	}
+	return s
+}
+
+func (c *TemporalWF) Init(metadata workflows.Metadata) error {
+	c.logger.Debugf("Temporal init start")
+	m, err := c.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	cOpt := client.Options{}
+	if m.HostPort != "" {
+		cOpt.HostPort = m.HostPort
+	}
+	if m.Identity != "" {
+		cOpt.Identity = m.Identity
+	}
+	// Create the workflow client
+	newClient, err := client.Dial(cOpt)
+	if err != nil {
+		return err
+	}
+	c.client = newClient
+
+	return nil
+}
+
+func (c *TemporalWF) Start(ctx context.Context, req *workflows.StartRequest) (*workflows.WorkflowReference, error) {
+	c.logger.Debugf("starting workflow")
+
+	if len(req.Options) == 0 {
+		c.logger.Debugf("no options provided")
+		return &workflows.WorkflowReference{}, errors.New("no options provided. At the very least, a task queue is needed")
+	}
+
+	if _, ok := req.Options["task_queue"]; !ok {
+		c.logger.Debugf("no task queue provided")
+		return &workflows.WorkflowReference{}, errors.New("no task queue provided")
+	}
+	taskQ := req.Options["task_queue"]
+
+	opt := client.StartWorkflowOptions{ID: req.WorkflowReference.InstanceID, TaskQueue: taskQ}
+	run, err := c.client.ExecuteWorkflow(ctx, opt, req.WorkflowName, req.Parameters)
+	if err != nil {
+		c.logger.Debugf("error when starting workflow: %v", err)
+		return &workflows.WorkflowReference{}, err
+	}
+	wfStruct := workflows.WorkflowReference{InstanceID: run.GetID()}
+	return &wfStruct, nil
+}
+
+func (c *TemporalWF) Terminate(ctx context.Context, req *workflows.WorkflowReference) error {
+	c.logger.Debugf("terminating workflow")
+
+	err := c.client.TerminateWorkflow(ctx, req.InstanceID, "", "")
+	if err != nil {
+		c.logger.Errorf("error when terminating workflow: %v", err)
```

Comment:
Now that I've had a bit more time to digest Go, I'm wondering if it would be more conventional to return a custom error instead of logging like this. The expectation is that the upstream caller takes care of logging for us:
```go
return fmt.Errorf("error terminating workflow: %w", err)
```
Same comment for all other places where we log and then return an error.

Reply:
I agree. Since `Terminate` is part of the Dapr runtime, the Runtime actually should log any error to the logger.

In this place we should just `return fmt.Errorf`